c7n/resources/route53.py

```diff
@@ -212,3 +214,107 @@ def process_resource_set(self, domains, keys):
             client.delete_tags_for_domain(
                 DomainName=d[self.id_key],
                 TagsToDelete=keys)
+
+
+@HostedZone.action_registry.register('enable-query-logging')
+class EnableQueryLogging(BaseAction):
+    """Enables query logging on a public hosted zone
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: enablednsquerylogging
+            resource: hostedzone
+            filters:
+                - type: query-logging-enabled
+                state: false
+            actions:
+                - type: enable-query-logging
+                state: true
+                logretentiondays: 30
+    """
+    permissions = ('route53:GetQueryLoggingConfig','route53:GetHostedZone',
+        'route53:CreateQueryLoggingConfig','route53:DeleteQueryLoggingConfig',
+        'logs:CreateLogGroup','logs:DescribeLogGroups','logs:PutRetentionPolicy')
+
+    schema = type_schema('enable-query-logging', state={'type': 'boolean'},
+        logretentiondays={'type': 'number'})
+
+    def process(self, resources):
+
+        client = local_session(self.manager.session_factory).client('route53')
+        state = self.data.get('state', False)
```

Comment:
we generally prefer to have actions which can enable/disable something default to not disabling, although thats occasionally context sensitive, generally the question on a default is would a user trying this action out in a policy find the default behavior confusing, and in this context disabling route53 logging without explicit config would be unintended/unexpected.